### PR TITLE
Test the encoding and use of fees; Upgrade bnsd

### LINKED ISSUE #686: Add support for fees
We have a fees field in all transactions, but I don't know if we actually use it when sending transactions.

We will soon (https://github.com/iov-one/weave/issues/264) start enforcing minimum fees server side and putting them in a "tip jar" Let's make sure we properly set the fees field if the client sends it (this can also be tested with current codebase... server handles it since May, just doesn't enforce a minimum).

Relevant code in weave if you want to look at it:

Implementation: https://github.com/iov-one/weave/blob/master/x/cash/decorator.go
Location in transaction: https://github.com/iov-one/weave/blob/master/cmd/bnsd/app/codec.proto#L29 (different field number, but same format as in 0.9)

### LINKED ISSUE #744: Test master using BNSd 0.11.x and Tendermint 0.29.x
Should work without code changes. @alpe do we have docker images ready to use for this setup?

### PR DESCRIPTION
**Pull request by webmaster128**
Closes #686
Closes #744

### CODE REVIEW

File: scripts/bnsd/bnsd_app.sh [hunk 1]
```diff
@@ -16,6 +16,7 @@ if [ ! -d "${BNSD_DIR}" ]; then
 fi
 
 exec docker run --user="$UID" \
+  --name "bnsd-app" \
```
Comment (ethanfrey) [inline scripts/bnsd/bnsd_app.sh]: add `--rm` there for better cleanup

File: scripts/bnsd/bnsd_tm.sh [hunk 2]
```diff
@@ -23,6 +23,7 @@ fi
 
 # tx indexing set in init
 exec docker run --user="$UID" \
+  --name "bnsd-tendermint" \
```
Comment (ethanfrey) [inline scripts/bnsd/bnsd_tm.sh]: As above, with `--rm`

File: scripts/bnsd/start.sh [hunk 3]
```diff
@@ -3,9 +3,9 @@ set -o errexit -o nounset -o pipefail
 command -v shellcheck > /dev/null && shellcheck "$0"
 
 # Choose from https://hub.docker.com/r/iov1/tendermint/tags/
-export BNSD_TM_VERSION=v0.27.4
+export BNSD_TM_VERSION=v0.29.1
```
Comment (ethanfrey) [inline scripts/bnsd/start.sh]: Nice

File: packages/iov-bns/src/bnsconnection.spec.ts [hunk 4]
```diff
@@ -459,6 +459,53 @@ describe("BnsConnection", () => {
...
     });
 
+    it("can send transaction with fees", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const chainId = connection.chainId();
+
+      const { profile, mainWalletId, faucet } = await userProfileWithFaucet(chainId);
+
+      const initialBalance = (await connection.getAccount({ pubkey: faucet.pubkey }))!.balance.find(
```
Comment (ethanfrey) [inline packages/iov-bns/src/bnsconnection.spec.ts]: nice one-liner

File: packages/iov-bns/src/encode.spec.ts [hunk 5]
```diff
@@ -138,15 +139,97 @@ describe("Encode", () => {
     expect(encoded.signature!.ed25519).toEqual(fromHex("aabbcc22334455"));
   });
 
+  describe("buildUnsignedTx", () => {
```
Comment (ethanfrey) [inline packages/iov-bns/src/encode.spec.ts]: good tests

File: scripts/bnsd/bnsd_app.sh (hunk 1 above)
Comment (webmaster128) [reply scripts/bnsd/bnsd_app.sh]: Hmm, I now changed the stop script to delete the container. We should be consistent. Any good reasons for one or the other solution?

Comment (webmaster128) [reply scripts/bnsd/bnsd_app.sh]: Postponed as this affects all containers we start: https://github.com/iov-one/iov-core/issues/748

**Pull request merged**